index.js

```diff
@@ -54,22 +54,26 @@ class LogForwardingPlugin {
     }
     const filterPattern = service.custom.logForwarding.filterPattern || '';
     const normalizedFilterID = !(service.custom.logForwarding.normalizedFilterID === false);
+    const roleArn = service.custom.logForwarding.roleArn || '';
     // Get options and parameters to make resources object
     const arn = service.custom.logForwarding.destinationARN;
     // Get list of all functions in this lambda
     const principal = `logs.${service.provider.region}.amazonaws.com`;
     // Generate resources object for each function
     // Only one lambda permission is needed
-    const resourceObj = {
-      LogForwardingLambdaPermission: {
-        Type: 'AWS::Lambda::Permission',
-        Properties: {
-          FunctionName: arn,
-          Action: 'lambda:InvokeFunction',
-          Principal: principal,
+    const resourceObj = {};
+    if (!roleArn) {
```

Comment:
if role arn isnt specified, resource object will be filled by the content below, the lambda log forwarding permission, if role arn is provided it will leave resource object empty since role arn and lambda forwarding permission aren't compatible parameters.